packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -750,6 +750,72 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
     });
   }
 
+  const aiPhoneCallSteps = steps.filter((s) => s.action === WorkflowActions.CAL_AI_PHONE_CALL && s.agentId);
+  if (aiPhoneCallSteps.length > 0) {
+    const aiService = createDefaultAIPhoneServiceProvider();
+    const externalToolErrors: string[] = [];
+
+    await Promise.all(
+      aiPhoneCallSteps.map(async (step) => {
+        if (!step.agentId) return;
+
+        try {
+          const agent = await ctx.prisma.agent.findUnique({
+            where: { id: step.agentId },
+            select: { providerAgentId: true },
+          });
+
+          if (!agent?.providerAgentId) {
+            log.error(`Agent not found for step ${step.id} agentId ${step.agentId}`);
+            return;
+          }
+
+          if (removedActiveOnIds.length > 0) {
+            try {
+              await aiService.removeToolsForEventTypes(agent.providerAgentId, removedActiveOnIds);
+            } catch (error) {
+              const message = `Failed to remove tools for removed event types from agent ${
+                agent.providerAgentId
+              }: ${error instanceof Error ? error.message : "Unknown error"}`;
+              externalToolErrors.push(message);
+              log.error(message);
+            }
+          }
+
+          if (newActiveOn.length > 0) {
+            await Promise.all(
+              newActiveOn.map(async (eventTypeId) => {
+                try {
+                  await aiService.updateToolsFromAgentId(agent.providerAgentId, {
```

Comment:
As we can't do thisyet  for future event types. Can remove the check box "Apply to all, including future event types" when cal.ai is selected as the workflow action